# Role-based setting of visibility (editing, use) for the form element is used

In case of a large number of roles in the configuration (from several
tens) it is not recommended to use the role-based visibility setting in
form elements (viewing and editing details by roles,
user visibility of form fields by roles, use of commands by
roles).

## Noncompliant Code Example

## Compliant Solution

## See

[Checking permissions](https://its.1c.ru/db/v8std#content:737:hdoc:1)