# Check if Use Always flag is enabled for the Reference attribute in dynamic list

In developing print commands (the Print subsystem of **Standard Subsystems Library**), the **Reference** attribute with the **Always Use** flag is required for these commands.

## Noncompliant Code Example

## Compliant Solution

## See

[Reference attribute in dynamic lists](https://support.1ci.com/hc/en-us/articles/360011004020-Reference-attribute-in-dynamic-lists)
 