# Check if User Visibility is disabled for the Ref field in dynamic list

To display additional attributes and info in the dynamic list forms of objects with properties, 
add the **Ref** field from the dynamic list query to the form table associated with this dynamic list. 
In this case, the added **Ref** field must be hidden from users by default by unchecking the **Visibility** check box in the **User visibility** property.

## Noncompliant Code Example

## Compliant Solution

## See

[Dynamic list settings](https://support.1ci.com/hc/en-us/articles/360014216499-Chapter-3-Setting-and-using-subsystems-upon-configuration-development)
[Reference attribute in dynamic lists](https://support.1ci.com/hc/en-us/articles/360011004020-Reference-attribute-in-dynamic-lists)
 