# Check if the Reference field is added to dynamic list

To display additional attributes and info in the list forms of objects with properties,
add the **Ref** field from the dynamic list query to the form table associated with this dynamic list.

## Noncompliant Code Example

## Compliant Solution

## See

[Dynamic list settings](https://support.1ci.com/hc/en-us/articles/360014216499-Chapter-3-Setting-and-using-subsystems-upon-configuration-development)
[Reference attribute in dynamic lists](https://support.1ci.com/hc/en-us/articles/360011004020-Reference-attribute-in-dynamic-lists)
 