# Check input field has correct list choice mode if choice list is not empty

Check input field has correct list choice mode if choice list is not empty.

## Noncompliant Code Example

## Compliant Solution

## See

