# Common cached modules.

Use the "ClientCached" postfixes for the modules that implement functions with repeated use of return values 
(upon the call or session time) on the client respectively.


## Noncompliant Code Example

## Compliant Solution

## See

[Common modules creating rules](https://support.1ci.com/hc/en-us/articles/360010988260-Common-modules-creating-rules)