# Common server modules for calling from the client contain server procedures and functions.

Name common server modules to be called from the client according to general rules of naming metadata objects. 
Make sure they include the "ServerCall" postfix.


## Noncompliant Code Example

## Compliant Solution

## See

[Common modules creating rules](https://support.1ci.com/hc/en-us/articles/360010988260-Common-modules-creating-rules)