# Using composite types like AnyRef

Do not use composite types AnyRef, CatalogRef, DocumentRef, and other for standard metadata objects stored in the infobase. 
Content of standard objects must be set explicitly.

## Noncompliant Code Example

## Compliant Solution

## See

