# Numeric data maximum length exceeded (31 symbols)

Application should equally work with all DBs, operation systems, web-browsers 
and various modes of work which 1C:Enterprise supports.

## Noncompliant Code Example

Maximum length of numeric data - 38 symbols

## Compliant Solution

[8.3. IBM Db2](https://its.1c.ru/db/v83doc#bookmark:dev:TI000001288)


> 8.3. Server IBM DB2
> ●  Maximum length of numeric data – 31 symbols (not 38).

## See

