# Check Russian Yo letter "ё" in the name, synonym or comment of metadata object

In Russian locale, it is not allowed to use the letter "ё" in names, synonyms and comments of metadata objects.


## Noncompliant Code Example

## Compliant Solution

## See

[Name, synonym, and comment (in Russian)](https://its.1c.ru/db/v8std#content:474:hdoc)
