# Accumulation or accounting register resource precision

Accumulation or accounting register resource precision must be no more than 25


## Noncompliant Code Example

## Compliant Solution

## See

