# Required role missing

In configuration should be declared three required roles:
1. FullAccess
2. SystemAdministrator
3. InteractiveOpenExternalReportsAndDataProcessors


## Noncompliant Code Example

## Compliant Solution

## See

