# The periodicity of execution a scheduled job is less than one minute.

The periodicity of execution a scheduled job is less than one minute.

## Noncompliant Code Example

## Compliant Solution

## See

