# The minimum job interval is 1 minute

The minimum job interval is 1 minute.

## Noncompliant Code Example

## Compliant Solution

## See

[Configuring scheduled job schedules](https://support.1ci.com/hc/en-us/articles/360011001440-Configuring-scheduled-job-schedules)
