# Document, which is meant to be posted, does not have privilege posoting mode on 

All the documents meant to be posted should have posting priviledge mode nad 
unposting privilege mode on.
Hence, no roles with extra rights to edit the registers should be created

## Noncompliant Code Example

## Compliant Solution

## See

