# Unsafe password storage

To minimize risks of unauthorized access to passwords, 
avoid storing passwords or any other confidential information in the infobase.

## Noncompliant Code Example

## Compliant Solution

## See
[Secure password storage](https://support.1ci.com/hc/en-us/articles/360011003100-Secure-password-storage)
