# Data constructor function return section

Check of module strict types system that return statements of function equals documentation comment return section types

## Noncompliant Code Example

## Compliant Solution

## See

