# Documentation comment Field has no type definition

Check of module strict types system that documentation comment field has section with types

## Noncompliant Code Example

## Compliant Solution

## See

