# Invocation parameter types intersects

Check of module strict types system that invocation expression has intercection with invocation parameter type

## Noncompliant Code Example

## Compliant Solution

## See

