# Referring to a non-existent role

In the parameters of the functions "IsInRole" and "Users.RolesAvailable" must
existing roles to be specified.

## Noncompliant Code Example


## Compliant Solution


## See
