# Method has to many parameters

It is not recommended to declare many parameters in methods (you need to focus on the number of no more than seven parameters),
while there should not be many parameters with default values (you need to focus on the number of no more than three such parameters). 
Otherwise, the readability of the calling code is greatly reduced. 
For example, you can easily make a mistake in the number of commas when passing optional parameters.

## Noncompliant Code Example

## Compliant Solution

## See

