# Checking the maximum number of empty rows

## Noncompliant Code Example

```bsl
Procedure Test()

    A1 = 1;
    
    
    A2 = 2;
    
EndProcedure
```

## Compliant Solution

```bsl
Procedure Test()

    A1 = 1;
    
    A2 = 2;
    
EndProcedure
```

## See

