# Self assigne variable

Assigning a variable to itself is meaningless and usually indicates an error.

## Noncompliant Code Example

Например, неправильно:

P1 = "";
P1 = P1;

## Compliant Solution

P1 = "";

