#  Unknown form parameter access

For parameterizable forms, all parameters must be provided in the form's parameter set. 
By this, all the form's parametrization options are described in the parameter set in a declarative way.

Specify the parameters in the code that opens the form (OpenForm).

## Noncompliant Code Example

## Compliant Solution

## См.

- [Opening form](https://kb.1ci.com/1C_Enterprise_Platform/Guides/Developer_Guides/1C_Enterprise_Development_Standards/Designing_user_interfaces/Implementation_of_form/Opening_forms/?language=en)